pkg/update/create.go

```diff
@@ -60,20 +62,20 @@ func getUpdateJobFromYaml() (*batchv1.Job, error) {
 	return udpateJob.(*batchv1.Job), nil
 }
 
-func constructUpdateJob(alluxio *alluxiov1alpha1.AlluxioCluster, update *alluxiov1alpha1.Update, updateJob *batchv1.Job) {
-	updateJob.Name = utils.GetUpdateJobName(update.Name)
-	updateJob.Namespace = alluxio.Namespace
+func constructUpdateJob(alluxio alluxiocluster.AlluxioClusterer, update Updater, updateJob *batchv1.Job) {
+	updateJob.Name = utils.GetUpdateJobName(update.GetName())
+	updateJob.Namespace = alluxio.GetNamespace()
 	var imagePullSecrets []corev1.LocalObjectReference
-	for _, secret := range alluxio.Spec.ImagePullSecrets {
+	for _, secret := range alluxio.ImagePullSecrets() {
 		imagePullSecrets = append(imagePullSecrets, corev1.LocalObjectReference{Name: secret})
 
 	}
 	updateJob.Spec.Template.Spec.ImagePullSecrets = imagePullSecrets
-	updateJob.Spec.Template.Spec.ServiceAccountName = alluxio.Spec.ServiceAccountName
-	updateJob.Spec.Template.Spec.Containers[0].Image = fmt.Sprintf("%s:%s", alluxio.Spec.Image, alluxio.Spec.ImageTag)
-	updateJob.Spec.Template.Spec.Containers[0].Command = []string{"go", "run", "/update.go", update.Spec.Path}
-	alluxioConfigMapName := utils.GetAlluxioConfigMapName(alluxio.Spec.NameOverride, alluxio.Name)
-	updateConfigMapName := utils.GetUpdateConfigmapName(alluxio.Spec.NameOverride, alluxio.Name)
+	updateJob.Spec.Template.Spec.ServiceAccountName = *alluxio.ServiceAccountName()
+	updateJob.Spec.Template.Spec.Containers[0].Image = fmt.Sprintf("%s:%s", *alluxio.Image(), *alluxio.ImageTag())
+	updateJob.Spec.Template.Spec.Containers[0].Command = []string{"go", "run", "/update.go", *update.Path()}
+	alluxioConfigMapName := utils.GetAlluxioConfigMapName(*alluxio.NameOverride(), alluxio.GetName())
+	updateConfigMapName := utils.GetUpdateConfigmapName(*alluxio.NameOverride(), alluxio.GetName())
```

Comment:
unfortunately, lots of checks missing assuming that all these fields are nil-able/optional